frontend/src/component/common/ConditionallyRender/ConditionallyRender.tsx

```diff
@@ -15,11 +15,12 @@ type TargetElement =
 
 type RenderFunc = () => JSX.Element;
 
-export const ConditionallyRender = ({
+export const ConditionallyRender: React.FC<IConditionallyRenderProps> = ({
+    children,
```

Comment:
This change is unrelated but allows us to use this syntax:
```typescript
            <ConditionallyRender condition={...}>
                <MyComponent />
            </ConditionallyRender>
```

Instead of this one which I find more awkward:
```typescript
            <ConditionallyRender condition={...} show={<MyComponent />} />
```

I don't have strong opinions on this change, so I'm happy to revert this (cc @FredrikOseberg as you may have some insights on whether this is a good change or not)

Reply:
Yes, it is. The else case can't be modeled easily, so this gives you some flexibility, you either use `show` with `elseShow` or you can also do:

```typescript
<ConditionallyRender condition={...} elseShow={<AnotherComponent />}>
    <MyComponent />
</ConditionallyRender>
```

Which I don't like that much. But for the sake of this PR I will split this apart.

Reply:
@gastonfournier 

I think we should keep consistency in the patterns we use, passing components as props for composition is quite common to do in React. I don't think we gain that much more clarity by adding another way to do the `show` part. If we do this I'd like to revisit how we do the other ConditionallyRenders in other parts of the application as well, but that is a much larger task.

Reply:
For an if (condition) -> component I like https://github.com/Unleash/unleash/pull/6223#discussion_r1487670255 better, but for else, I don't so much. I think it makes things more readable, getting read of the property and using children, is more html-like. But I reverted this change for the sake of this feature :)